File: Leetcode/2739. Total Distance Traveled.cpp
// https://leetcode.com/problems/total-distance-traveled/

class Solution {
public:
    int distanceTraveled(int mainTank, int additionalTank) {
        int result = 0;
        while (mainTank >= 5 && additionalTank > 0) {
            int transfered = min(int(mainTank / 5), additionalTank);
            result += 50 * transfered;
            mainTank -= 4 * transfered;
            additionalTank -= transfered;
        }
        return result + mainTank * 10;
    }
};

File: Leetcode/2739. Total Distance Traveled.py
# https://leetcode.com/problems/total-distance-traveled/

class Solution:
    def distanceTraveled(self, mainTank: int, additionalTank: int) -> int:
        result = 0
        while mainTank >= 5 and additionalTank > 0:
            transfered = min(mainTank // 5, additionalTank)
            result += 10 * transfered * 5
            mainTank -= 4 * transfered
            additionalTank -= transfered
        result += mainTank * 10
        return result
